src/Stack/Setup.hs

```diff
@@ -140,6 +140,8 @@ data SetupOpts = SetupOpts
     , soptsUpgradeCabal :: !Bool
     -- ^ Upgrade the global Cabal library in the database to the newest
     -- version. Only works reliably with a stack-managed installation.
+    , soptsInstallCabal :: !(Maybe Version)
+    -- ^ Install a specific version of Cabal.
```

Comment:
This sounds like a good idea. I haven't seen an 'optional argument' but will have a look and see what is possible.